cdc/processor/processor.go

```diff
@@ -77,7 +77,7 @@ type processor struct {
 
 	lazyInit            func(ctx cdcContext.Context) error
 	createTablePipeline func(ctx cdcContext.Context, tableID model.TableID, replicaInfo *model.TableReplicaInfo) (pipeline.TablePipeline, error)
-	newAgent            func(ctx cdcContext.Context) (scheduler.Agent, error)
+	newAgent            func(cdcContext.Context, *model.Liveness) (scheduler.Agent, error)
```

Comment:
Can you consider making an `AgentFactory` interface? So that we can put whatever extra arguments we need into the implementation of an `AgentFactory`, so that further changes will not affect how the interface method `NewAgent` is called.

Just a suggestion. If you think this part of code is not subject to anymore significant changes, I think the current way is fine.

Reply:
I don't think it will change anytime soon, let's add the factory when necessary.